### ISSUE TITLE
Continuous integration/deployment plans?

### ISSUE THREAD

**Issue by andrew-gardener** [tl-its-umich-edu/my-learning-analytics#382]
I'm wondering if there are any current or future plans around CI/CD in this project?

For example, I'd be happy to whip up a Travis CI config for pushing docker images to docker hub for you if your interested (though I suppose it will depend on the License decision).

Also a side question is what are your thoughts on automated testing for this project? Currently there doesn't seem to be any but I assume that's due to trying to rapid development/prototype? I can begin to add unit tests to some of my PRs if there is interest in starting to add automated tests

**Comment by jonespm** [comment]
Currently we're using Openshift for CI/CD locally. I'm not sure if there's value at the moment for Docker images to be pushed to Docker Hub. The software at the moment cannot be run even in a demo mode without configuration. 

We are interested in automated testing and hope to get some student help this summer on that. I'm not entirely sure how much can be tested since so much of the application is D3 based and typical tools like Selenium don't interact with SVG. I saw[ threads like this](https://stackoverflow.com/questions/48486980/d3-automation-testing) for other tools to simulate mouse clicks. If we moved to Angular then there are probably tools for that too. 

I think it would be great to have unit test and tests for the apis but that is going to require adding sample data to run tests against. So yeah we'd love all of these things but they each have dependencies. @jennlove-um was talking about wanted to get another meeting setup so we can find out what everyone's working on.
